base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -847,17 +875,13 @@ impl LMDBDatabase {
             total_utxo_sum = &total_utxo_sum + &output.commitment;
             output_mmr.push(output.hash())?;
             witness_mmr.push(output.witness_hash())?;
-            trace!(
-                target: LOG_TARGET,
-                "Inserting output `{}`",
-                to_hex(&output.commitment.as_bytes())
-            );
+            trace!(target: LOG_TARGET, "Inserting output `{}`", output.commitment.to_hex());
             self.insert_output(
                 txn,
                 block_hash.clone(),
                 header.height,
                 output,
-                (witness_mmr.get_leaf_count()? - 1) as u32,
+                (output_mmr.get_leaf_count() - 1) as u32,
```

Comment:
Just ensure these will always return the same count. I am not sure if output.mmr will return a smaller number if it actually gets pruned.  And witness_mmr.size() == output_mmr.size()

Reply:
Yeah they do, not sure why I used `witness_mmr` in the first place.

Reply:
`output_mmr.len()` will give you the pruned length (artificially created by taking `leaf_count - cardinality(deleted)`)
but if we ever decide to prune the MMR we could run into trouble. So I think I should revert this change.